tests/behaviour/steps/lib.rs

```diff
@@ -51,18 +54,19 @@ mod thing_util {
 
 #[derive(Debug, Default, World)]
 pub struct Context {
-    server_dir: Option<TempDir>,
-    server: Option<typedb::Server>,
+    server: Option<Arc<Mutex<typedb::Server>>>,
     active_transaction: Option<ActiveTransaction>,
 
     objects: HashMap<String, Option<ObjectWithKey>>,
     object_lists: HashMap<String, Vec<Object<'static>>>,
     attributes: HashMap<String, Option<Attribute<'static>>>,
     attribute_lists: HashMap<String, Vec<Attribute<'static>>>,
+
+    clean_databases_after: bool,
 }
 
 impl Context {
-    pub async fn test(glob: &'static str) -> bool {
+    pub async fn test(glob: &'static str, clean_databases_after: bool) -> bool {
```

Comment:
Could make 2 public functions with or without cleanup if you wanted. It's fine to have a set of boolean flags now, can become a config in the future.